cdk/lib/prism.ts

```diff
@@ -48,27 +49,33 @@ export class PrismStack extends GuStack {
     });
 
     const distBucket: string = this.getParam(GuDistributionBucketParameter.parameterName).valueAsString;
-    const s3Key = [distBucket, this.stack, this.stage, this.app, `${this.app}.deb`].join("/");
 
-    // TODO move to UserData.forLinux
-    const userData = `#!/bin/bash -ev
-
-      aws --region ${this.region} s3 cp s3://${s3Key} /tmp/
-      dpkg -i /tmp/prism.deb`;
+    const userData = new GuUserData(this, {
+      distributable: {
+        bucketName: distBucket,
+        fileName: "prism.deb",
+        executionStatement: `dpkg -i /${this.app}/prism.deb`,
+      },
+    });
 
     const asg = new GuAutoScalingGroup(this, "AutoscalingGroup", {
       overrideId: true,
       vpc,
       vpcSubnets: { subnets },
       role: role,
-      userData: userData,
-      minCapacity: 2,
-      maxCapacity: 4,
-      desiredCapacity: 2,
+      userData: userData.userData,
+      stageDependentProps: {
+        [Stage.CODE]: {
+          minimumInstances: 2,
```

Comment:
I wanted to preserve the existing configuration for the upgrade, but perhaps we should run a single instance in `CODE`. I guess it'll save a little bit of money.

Reply:
https://github.com/guardian/prism/pull/137